frontend/src/hooks/useKeycloakWrapper.tsx

```diff
@@ -1,60 +1,140 @@
 import { useKeycloak } from '@react-keycloak/web';
-import { SYSTEM_ADMINISTRATOR, AGENCY_ADMINISTRATOR } from 'constants/strings';
+import { Roles } from 'constants/roles';
 
+/**
+ * IUserInfo interface, represents the userinfo provided by keycloak.
+ */
 export interface IUserInfo {
+  displayName?: string;
   username: string;
   name?: string;
   preferred_username?: string;
   firstName?: string;
   lastName?: string;
   email: string;
-  Groups: any[];
+  groups: string[];
+  roles: string[];
   given_name?: string;
   family_name?: string;
   agencies: number[];
 }
 
-function useKeycloakWrapper() {
+/**
+ * IKeycloak interface, represents the keycloak object for the authenticated user.
+ */
+interface IKeycloak {
+  obj: any;
+  displayName?: string;
+  username: string;
+  name?: string;
+  preferred_username?: string;
+  firstName?: string;
+  lastName?: string;
+  email?: string;
+  roles: string[];
+  agencyId?: number;
+  isAdmin: boolean;
+  hasRole(role?: string | Array<string>): boolean;
+  hasClaim(claim?: string | Array<string>): boolean;
+  hasAgency(agency?: number): boolean;
+}
+
+/**
+ * Provides extension methods to interact with the `keycloak` object.
+ */
+function useKeycloakWrapper(): IKeycloak {
   const { keycloak } = useKeycloak();
-  const hasClaim = (claim?: string) => {
-    return !claim || keycloak?.hasResourceRole(claim);
-  };
   const userInfo = keycloak?.userInfo as IUserInfo;
-  const hasRole = (role?: string) => {
-    return !role || userInfo?.Groups?.includes(role);
+
+  /**
+   * Determine if the user has the specified 'claim'
+   * @param claim - The name of the claim
+   */
+  const hasClaim = (claim?: string | Array<string>): boolean => {
+    return (
+      !claim ||
+      (typeof claim === 'string'
+        ? userInfo?.roles?.includes(claim)
+        : claim.some(c => userInfo?.roles?.includes(c)))
+    );
+  };
+
+  /**
+   * Determine if the user belongs to the specified 'role'
+   * @param role - The role name or an array of role name
+   */
+  const hasRole = (role?: string | Array<string>): boolean => {
+    return (
+      !role ||
+      (typeof role === 'string'
+        ? userInfo?.groups?.includes(role)
+        : role.some(r => userInfo?.groups?.includes(r)))
+    );
+  };
+
+  /**
+   * Determine if the user belongs to the specified 'agency'
+   * @param agency - The agency name
+   */
+  const hasAgency = (agency?: number): boolean => {
+    return !agency || userInfo?.agencies?.includes(agency);
   };
-  const roles = () => {
-    return userInfo?.Groups ? [...userInfo?.Groups] : [];
+
+  /**
+   * Return an array of roles the user belongs to
+   */
+  const roles = (): Array<string> => {
+    return userInfo?.groups ? [...userInfo?.groups] : [];
```

Comment:
Make sure you check for case-sensitivity here. Again, what I've seen in the JSON response is `obj.Groups`